perfkitbenchmarker/virtual_machine.py

```diff
@@ -138,6 +150,9 @@ def __init__(self, vm_spec):
     self._total_memory_kb = None
     self._num_cpus = None
 
+    self._has_remote_command_script_lock = threading.Lock()
```

Comment:
threading.lock() returns a lock object, not a boolean. Please consider renaming this to _remote_command_script_lock (basically, drop the "has" prefix which implies a boolean).